flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/RecordWriter.java

```diff
@@ -164,7 +164,7 @@ public BufferConsumer broadcastEvent(AbstractEvent event) throws IOException {
 			if (flushAlways) {
 				flushAll();
 			}
-			return eventBufferConsumer;
```

Comment:
You don't need to close the `eventBufferConsumer` since the try-with-resources should already do that. Returning the closed value, however, is kind of strange and only needed in one test. Let's make the method return `void` as you suggested and adapt the test.